src/posix/http_client_curl.cc

```diff
@@ -41,12 +44,15 @@ std::string RunCurl(const std::string& cmd) {
   FILE* pipe = ::popen(cmd.c_str(), "r");
   assert(pipe);
   char s[8 * 1024];
-  ::fgets(s, sizeof(s) / sizeof(s[0]), pipe);
+  std::string result;
+  while (nullptr != ::fgets(s, sizeof(s) / sizeof(s[0]), pipe)) {
+    result += s;
```

Comment:
Wow, why do you think that fgets would not get the last line without \n? Never heard it before. Is it a documented behavior?

> On 15.3.2015, at 00.00, Dima <notifications@github.com> wrote:
> 
> In src/posix/http_client_curl.cc:
> 
> > @@ -41,12 +44,15 @@ std::string RunCurl(const std::string& cmd) {
> >    FILE* pipe = ::popen(cmd.c_str(), "r");
> >    assert(pipe);
> >    char s[8 * 1024];
> > -  ::fgets(s, sizeof(s) / sizeof(s[0]), pipe);
> > +  std::string result;
> > +  while (nullptr != ::fgets(s, sizeof(s) / sizeof(s[0]), pipe)) {
> > +    result += s;
> If the last line does not have a newline character, it will not be there in the response of fgets(). Which, for my taste, is inconsistent.
> 
> —
> Reply to this email directly or view it on GitHub.
>